pkg/registries/registries.go

```diff
@@ -155,10 +200,30 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Blocked = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, blockedReg) {
+			scope := reg.Location
+			// Set scope to Prefix if it exists
+			if reg.Prefix != "" {
+				scope = reg.Prefix
+			}
+			if scopeMatchesRegistry(scope, blockedReg) || isSubScopeOfReg(scope, blockedReg) {
 				reg.Blocked = true
 			}
 		}
 	}
 	return nil
 }
+
+// IsValidWildcard returns true if scope is a valid wildcard entry such as "*.registry.com"
+// This function can be used to validate the registries entries priori to calling EditRegistriesConfig
+// in the MCO or builds code
+func IsValidWildcard(scope string) bool {
+	if strings.Contains(scope, "*") && !strings.HasPrefix(scope, "*.") {
+		return false
+	}
+
+	// Check that the prefix is only "*."" and doesn't contain any /,@,: characters
+	if strings.HasPrefix(scope, "*.") && !strings.ContainsAny(scope, "/@:") {
+		return true
+	}
```

Comment:
(Non-blocking: This might be easier to follow:
```go
if !strings.HasPrefix(scope, "*.") { return false } // Not a wildcard - or add validation for non-wildcards here
if strings.ContainsAny(scope[2:], "/@:*") { return false } // Invalid content
return true
```
or maybe some transformation of this WRT if/else negation: 
- We only need to check for the prefix once.
- This naturally handles the special case in `*`
- (No need to check the the `*.` prefix for invalid characters we know aren’t present.)

)

Reply:
Fixed.